### ISSUE TITLE
[1.12.2] The "Heart Outline" doesn't render if you have more than half a heart on the current row. It also doesn't render at all when Vanilla's Heart Render is disabled.

### ISSUE THREAD

**Issue by SonicX8000** [SilentChaos512/ScalingHealth#129]
Minecraft: 1.12.2
Forge: 14.23.4.2750

Mods:
ScalingHealth-1.12.2-1.3.24-119
SilentLib-1.12.2-2.3.17-138

Image:
![](https://cdn.discordapp.com/attachments/377171551155650560/479237493322153984/whiteoutlinebugx.PNG)

-----
When you have Half a Heart on the current row of hearts the White Outline will render... but it will disappear when you have more than Half a Heart.

If Vanilla Hearts are disabled the White Outline will not render at all no matter what.

**Comment by SonicX8000** [comment]
![](https://cdn.discordapp.com/attachments/377171551155650560/479637075814252544/heartoutlineallrows.PNG)

The white heart outline shows on All Rows now instead of the Highest Row.

Also... while unrelated to the current issue... I think the counter for "HEALTH_AND_MAX" isn't working correctly as it tells me I have 0/70 Health but the HUD shows Half a Heart. Unless I have like maybe 0.02 health left or something... don't think I ever seen this before either.

**Comment by SilentChaos512** [comment]
I imagine the 0/70 is just a rounding issue. A half heart will render if
you have even a sliver of health, like you said. But I'll take a look at
the exact values.
I also changed the way poison/wither is displayed. If you have time to
check those out, please let me know what you think.

**Comment by SonicX8000** [comment]
Ooo... tiny hearts much like how the absorb hearts/shields... thing is... since the hearts are... well... heart shaped it was kinda tricky to work with. Anyways...

![](https://cdn.discordapp.com/attachments/377171551155650560/479699263086002187/poisonwitheroverlay.PNG)

I gave the tiny hearts the black outline much like with the absorb hearts and I changed the coloring on them since I kinda feel that the current colors would've blended in with the custom outline color. The current poison overlay would've blended in with the green hearts and the wither hearts would've blended in on the first row of hearts.

Using the vanilla status icons just to get the palette colors... I went with a greenish olive color for poison and a dark palish pink color for wither. I was trying to see if I could have some sort of icon over the hearts like a poison droplet for poison, a skull for wither but the small size of the hearts... yea.
